src/workers/personalizedDigestEmail.ts

```diff
@@ -129,10 +132,59 @@ const digestTypeToFunctionMap: Record<
   },
   [UserPersonalizedDigestType.ReadingReminder]: async (data, con) => {
     const { personalizedDigest, emailSendTimestamp, deduplicate = true } = data;
-    const emailSendDate = new Date(emailSendTimestamp);
+    const notificationSendTimestamp = new Date(emailSendTimestamp);
+    const currentDate = new Date();
+    await dedupedSend(
+      () =>
+        sendReadingReminderPush(
+          [personalizedDigest.userId],
+          notificationSendTimestamp,
+        ),
+      {
+        con,
+        personalizedDigest,
+        date: currentDate,
+        deduplicate,
+      },
+    );
+  },
+  [UserPersonalizedDigestType.StreakReminder]: async (data, con, logger) => {
+    const { personalizedDigest, emailSendTimestamp, deduplicate = true } = data;
+    const { userId } = personalizedDigest;
+
+    const notificationSendTimestamp = new Date(emailSendTimestamp);
     const currentDate = new Date();
+    const userSettings = await con
+      .getRepository(Settings)
+      .findOneBy({ userId });
+
+    // Safety measure to prevent sending streak reminders to users who have opted out
+    // but for some reason still have a streak reminder scheduled
+    if (userSettings?.optOutReadingStreak) {
+      return;
+    }
+
+    const userStreak = await con
+      .getRepository(UserStreak)
+      .findOneBy({ userId });
+
+    if (!userStreak) {
```

Comment:
Maybe we should not send too when the streak count is 0.

Reply:
Added a check for it :)

Reply:
@omBratteng Do we check here if you already read today?

Reply:
@rebelchris yes. We check if the day of current time and lastViewAt is the same here
https://github.com/dailydotdev/daily-api/blob/afacdd327e2127f649e1e4e06ceea395f2952cbc/src/workers/personalizedDigestEmail.ts#L179